go/vt/mysqlctl/azblobbackupstorage/azblob.go

```diff
@@ -0,0 +1,297 @@
+/*
+Copyright 2020 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package azblobbackupstorage implements the BackupStorage interface
+// for Azure Blob Storage
+package azblobbackupstorage
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"vitess.io/vitess/go/vt/concurrency"
+	"vitess.io/vitess/go/vt/log"
+	"vitess.io/vitess/go/vt/mysqlctl/backupstorage"
+)
+
+var (
+	// This is the account name
+	accountName = flag.String("azblob_backup_account_name", "", "Azure Account Name for Backups")
+
+	// This is the private access key
+	accountKey = flag.String("azblob_backup_account_key", "", "Azure Account Key")
+
+	// This is the name of the container that will store the backups
+	containerName = flag.String("azblob_backup_contain_name", "", "Azure Blob Contain Name")
+
+	// This is an optional previx to prepend to all files
+	prefix = flag.String("azblob_backup_prefix", "", "Azure Blob prefix")
+
+	azBlobParallelism = flag.Int("azblob_parallelism", 1, "Azure blob operation parallelism (requires extra memory when increased)")
```

Comment:
Since this flag only affects this package, not all places that may call the azblob client, the name should be scoped more specifically:

```suggestion
	azBlobParallelism = flag.Int("azblob_backup_parallelism", 1, "Azure blob operation parallelism (requires extra memory when increased)")
```